### PR TITLE
Add "Last heard" relative-time data point above map on QSO Details sheet

### PR DESCRIPTION
**Pull request by patrickrb**
## Summary
Adds a **Last heard** data point to the QSO Details bottom sheet, rendered as a relative "ago" time (`just now`, `45s ago`, `12m ago`, `3h ago`, `2d ago`) positioned **above the map**. Operators get immediate recency context without mentally computing elapsed time from a raw timestamp.

## What changed
- **`QsoSheet.kt`**
  - New pure, unit-tested `computeLastHeard(utcTimeMillis, nowMillis): LastHeard` that buckets elapsed time into a `LastHeard` sealed type (`Unknown`/`JustNow`/`Seconds`/`Minutes`/`Hours`/`Days`). Kept Compose- and resource-free so it's directly testable and localizable.
  - New thin `LastHeardRow` composable that maps a bucket to a localized string via `lastHeardText` and renders the label/value row above `QsoPathMap`.
  - `QsoSheetContent` now observes the app's existing `MainViewModel.timerSec` (~1s UTC heartbeat) and passes it in, so the value refreshes while the sheet is open.
- **`strings_compose.xml`** — new `qso_last_heard*` strings (label + `just now` / `%1$ds ago` / `%1$dm ago` / `%1$dh ago` / `%1$dd ago` / `--` placeholder).
- **`LastHeardTest.kt`** — 7 tests covering every bucket boundary plus the edge cases.

## Acceptance criteria
- ✅ **Last heard** label/value shown above the map.
- ✅ Relative format with `ago` suffix.
- ✅ Updates as time passes (driven by `timerSec`, the same clock the decode list uses).
- ✅ Edge cases handled cleanly: missing/non-positive timestamp → `--`; future timestamp (clock skew) clamps to `just now` (never negative); very old → plain day count (e.g. `500d ago`).

## Notes / assumptions
- Thresholds (5 s / 60 s / 60 m / 24 h) mirror the existing decode-row `formatTimeAgo` so recency reads consistently across the app.
- Timestamp source is `Ft8Message.utcTime` (epoch ms); "now" is `UtcTimer`-based `timerSec`, both already used elsewhere.
- New strings are English-only for now (other locales fall back to the default resource); no lint enforcement of `MissingTranslation` in this module.

## Testing
`./gradlew :app:testDebugUnitTest --tests '*.LastHeardTest' --tests '*.QsoSheetLogicTest'` → BUILD SUCCESSFUL (7/7 new tests pass; existing sheet tests unaffected). Main source compiles.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: ft8af/app/src/main/kotlin/radio/ks3ckc/ft8af/ui/decode/QsoSheet.kt [hunk 1]
```diff
@@ -112,6 +112,8 @@ private fun QsoSheetContent(
     val isActivated by mainViewModel.ft8TransmitSignal.mutableIsActivated.observeAsState(false)
     val toCallsign by mainViewModel.ft8TransmitSignal.mutableToCallsign.observeAsState()
     val txFunctionOrder by mainViewModel.ft8TransmitSignal.mutableFunctionOrder.observeAsState(6)
+    // App-wide UTC clock (~1s heartbeat); keeps the "Last heard" value fresh.
+    val nowMillis by mainViewModel.timerSec.observeAsState(UtcTimer.getSystemTime())
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8af/app/src/main/kotlin/radio/ks3ckc/ft8af/ui/decode/QsoSheet.kt]: Valid — good catch. `timerSec` changes ~1×/s and the sheet also hosts `QsoPathMap` (which redraws on every recompose), so observing the tick at the sheet level repainted the whole map each second. Fixed in e2fcbb13: the `observeAsState(timerSec)` now lives inside `LastHeardRow`, so only that row recomposes per tick.

File: ft8af/app/src/main/kotlin/radio/ks3ckc/ft8af/ui/decode/QsoSheet.kt [hunk 2]
```diff
@@ -813,6 +820,94 @@ private fun CurrentTxBanner(
...
+/**
+ * Bucket the elapsed time between [utcTimeMillis] (when the station was heard)
+ * and [nowMillis] into a [LastHeard] value. Edge cases handled cleanly:
+ *  - a missing/non-positive timestamp -> [LastHeard.Unknown]
+ *  - a future timestamp (clock skew) -> [LastHeard.JustNow] (never negative)
+ *  - a very old timestamp -> a plain [LastHeard.Days] count
+ *
+ * Thresholds mirror the existing decode-row "ago" formatter (5 s / 60 s / 60 m /
+ * 24 h) so recency reads consistently across the app.
+ */
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8af/app/src/main/kotlin/radio/ks3ckc/ft8af/ui/decode/QsoSheet.kt]: Fair — the wording does differ ("just now" vs the decode row's "now"). The intent was only to mirror the *thresholds*, not the exact text. Reworded the KDoc in e2fcbb13 to say so and to point at `lastHeardText` as the source of the display wording.

File: ft8af/app/src/test/kotlin/radio/ks3ckc/ft8af/ui/decode/LastHeardTest.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    @Test
+    fun `hours bucket covers one hour up to under a day`() {
+        assertThat(computeLastHeard(now - 3_600_000L, now)).isEqualTo(LastHeard.Hours(1))
+        assertThat(computeLastHeard(now - 3L * 3_600_000L, now)).isEqualTo(LastHeard.Hours(3))
+        assertThat(computeLastHeard(now - (86_400_000L - 1L), now)).isEqualTo(LastHeard.Hours(23))
+    }
+
+    @Test
+    fun `days bucket covers a day and beyond, including very old timestamps`() {
+        assertThat(computeLastHeard(now - 86_400_000L, now)).isEqualTo(LastHeard.Days(2 - 1))
```
[the remark this answers is not in the document]
Comment (patrickrb) [reply ft8af/app/src/test/kotlin/radio/ks3ckc/ft8af/ui/decode/LastHeardTest.kt]: Agreed — changed to assert `LastHeard.Days(1)` literally in e2fcbb13.
